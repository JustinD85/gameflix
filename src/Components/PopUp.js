import React, { Component } from 'react';

export default class PopUp extends Component {
  render() {
    const { isSearch, game, closePopUp } = this.props;
    let className;
    let iconClassName;

    isSearch ? className = "search-popup" : className = "PopUp"

    switch (game.challenge_level) {
      case 'simple':
        iconClassName = "fas fa-thermometer-quarter"
        break;
      case 'average':
        iconClassName = "fas fa-thermometer-half"
        break;
      default:
        iconClassName = "fas fa-thermometer-full"
    }

    return (
      <div className={className}>
        <section class="game-details">

          <i onClick={closePopUp} className="fas fa-times"></i>
          <h1 class="game-name">
            {game.game}
          </h1>
          <div class="game-info">
            <p class="age-range"><i class="fas fa-birthday-cake"></i> Ages {game.min_age} +</p>
            <p class="num-of-players"><i class="fas fa-users"></i> {game.min_players} - {game.max_players} players</p>
            <p class="num-of-minutes"><i class="fas fa-clock"></i> {game.number_of_minutes} min.</p>
          </div>
          <p class="game-description">{game.description}</p>
          <p class="difficulty-level">
            Challenge level: {game.challenge_level}
            <i className={iconClassName}></i>
          </p>
          <img className="carousel-image popup-image" src={game.img} alt="game in box" />
        </section>

        <section class="youtube-video">
          <iframe
            width="100%"
            height="100%"
            src={`https://www.youtube.com/embed/${game.youtube}?autoplay=0`}
            frameBorder="0"
            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen
            title="featured-video">
          </iframe>
        </section>
      </div>
    )
  }
}

