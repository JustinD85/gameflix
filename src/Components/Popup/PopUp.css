.Popup {
  height: 500px;
}