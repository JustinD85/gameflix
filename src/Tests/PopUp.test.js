import React from 'react';
import { shallow } from 'enzyme';
import PopUp from '../Components/PopUp';

describe('PopUp', () => {
  let wrapper;

  const game = {
    "game": "Don't Be A Loser",
    "description": "Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!",
    "min_age": 13,
    "min_players": 4,
    "max_players": 10,
    "number_of_minutes": "30",
    "genre_ID": [6, 11, 13],
    "expansion_pack": false ,
    "challenge_level": "average",
    "publisher": "All Over The Board",
    "youtube": "ZYvvhOzYx_4",
    "img": "https://cdn.shopify.com/s/files/1/1911/5793/products/DBAL-main-photo-box_350x.jpg?v=1512623523"
  }

  beforeEach(() => {
    wrapper = shallow(
      <PopUp  game={game} />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }) 

})