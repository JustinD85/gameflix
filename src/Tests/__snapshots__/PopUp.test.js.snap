// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PopUp
    game={
      Object {
        "challenge_level": "average",
        "description": "Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!",
        "expansion_pack": false,
        "game": "Don't Be A Loser",
        "genre_ID": Array [
          6,
          11,
          13,
        ],
        "img": "https://cdn.shopify.com/s/files/1/1911/5793/products/DBAL-main-photo-box_350x.jpg?v=1512623523",
        "max_players": 10,
        "min_age": 13,
        "min_players": 4,
        "number_of_minutes": "30",
        "publisher": "All Over The Board",
        "youtube": "ZYvvhOzYx_4",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          class="game-details"
        >
          <i
            className="fas fa-times"
          />
          <h1
            class="game-name"
          >
            Don't Be A Loser
          </h1>
          <div
            class="game-info"
          >
            <p
              class="age-range"
            >
              <i
                class="fas fa-birthday-cake"
              />
               Ages 
              13
               +
            </p>
            <p
              class="num-of-players"
            >
              <i
                class="fas fa-users"
              />
               
              4
               - 
              10
               players
            </p>
            <p
              class="num-of-minutes"
            >
              <i
                class="fas fa-clock"
              />
               
              30
               min.
            </p>
          </div>
          <p
            class="game-description"
          >
            Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!
          </p>
          <p
            class="difficulty-level"
          >
            Challenge level: 
            average
            <i
              className="fas fa-thermometer-half"
            />
          </p>
          <img
            alt="game in box"
            className="carousel-image"
            src="https://cdn.shopify.com/s/files/1/1911/5793/products/DBAL-main-photo-box_350x.jpg?v=1512623523"
          />
        </section>,
        <section
          class="youtube-video"
        >
          <iframe
            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen={true}
            frameBorder="0"
            height="100%"
            src="https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0"
            title="featured-video"
            width="100%"
          />
        </section>,
      ],
      "className": "PopUp",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-times"
            />,
            <h1
              class="game-name"
            >
              Don't Be A Loser
            </h1>,
            <div
              class="game-info"
            >
              <p
                class="age-range"
              >
                <i
                  class="fas fa-birthday-cake"
                />
                 Ages 
                13
                 +
              </p>
              <p
                class="num-of-players"
              >
                <i
                  class="fas fa-users"
                />
                 
                4
                 - 
                10
                 players
              </p>
              <p
                class="num-of-minutes"
              >
                <i
                  class="fas fa-clock"
                />
                 
                30
                 min.
              </p>
            </div>,
            <p
              class="game-description"
            >
              Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!
            </p>,
            <p
              class="difficulty-level"
            >
              Challenge level: 
              average
              <i
                className="fas fa-thermometer-half"
              />
            </p>,
            <img
              alt="game in box"
              className="carousel-image"
              src="https://cdn.shopify.com/s/files/1/1911/5793/products/DBAL-main-photo-box_350x.jpg?v=1512623523"
            />,
          ],
          "class": "game-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-times",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Don't Be A Loser",
              "class": "game-name",
            },
            "ref": null,
            "rendered": "Don't Be A Loser",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  class="age-range"
                >
                  <i
                    class="fas fa-birthday-cake"
                  />
                   Ages 
                  13
                   +
                </p>,
                <p
                  class="num-of-players"
                >
                  <i
                    class="fas fa-users"
                  />
                   
                  4
                   - 
                  10
                   players
                </p>,
                <p
                  class="num-of-minutes"
                >
                  <i
                    class="fas fa-clock"
                  />
                   
                  30
                   min.
                </p>,
              ],
              "class": "game-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      class="fas fa-birthday-cake"
                    />,
                    " Ages ",
                    13,
                    " +",
                  ],
                  "class": "age-range",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "class": "fas fa-birthday-cake",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " Ages ",
                  13,
                  " +",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      class="fas fa-users"
                    />,
                    " ",
                    4,
                    " - ",
                    10,
                    " players",
                  ],
                  "class": "num-of-players",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "class": "fas fa-users",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                  4,
                  " - ",
                  10,
                  " players",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      class="fas fa-clock"
                    />,
                    " ",
                    "30",
                    " min.",
                  ],
                  "class": "num-of-minutes",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "class": "fas fa-clock",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " ",
                  "30",
                  " min.",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!",
              "class": "game-description",
            },
            "ref": null,
            "rendered": "Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Challenge level: ",
                "average",
                <i
                  className="fas fa-thermometer-half"
                />,
              ],
              "class": "difficulty-level",
            },
            "ref": null,
            "rendered": Array [
              "Challenge level: ",
              "average",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-thermometer-half",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "game in box",
              "className": "carousel-image",
              "src": "https://cdn.shopify.com/s/files/1/1911/5793/products/DBAL-main-photo-box_350x.jpg?v=1512623523",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <iframe
            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
            allowFullScreen={true}
            frameBorder="0"
            height="100%"
            src="https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0"
            title="featured-video"
            width="100%"
          />,
          "class": "youtube-video",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "allow": "accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture",
            "allowFullScreen": true,
            "frameBorder": "0",
            "height": "100%",
            "src": "https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0",
            "title": "featured-video",
            "width": "100%",
          },
          "ref": null,
          "rendered": null,
          "type": "iframe",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            class="game-details"
          >
            <i
              className="fas fa-times"
            />
            <h1
              class="game-name"
            >
              Don't Be A Loser
            </h1>
            <div
              class="game-info"
            >
              <p
                class="age-range"
              >
                <i
                  class="fas fa-birthday-cake"
                />
                 Ages 
                13
                 +
              </p>
              <p
                class="num-of-players"
              >
                <i
                  class="fas fa-users"
                />
                 
                4
                 - 
                10
                 players
              </p>
              <p
                class="num-of-minutes"
              >
                <i
                  class="fas fa-clock"
                />
                 
                30
                 min.
              </p>
            </div>
            <p
              class="game-description"
            >
              Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!
            </p>
            <p
              class="difficulty-level"
            >
              Challenge level: 
              average
              <i
                className="fas fa-thermometer-half"
              />
            </p>
            <img
              alt="game in box"
              className="carousel-image"
              src="https://cdn.shopify.com/s/files/1/1911/5793/products/DBAL-main-photo-box_350x.jpg?v=1512623523"
            />
          </section>,
          <section
            class="youtube-video"
          >
            <iframe
              allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
              allowFullScreen={true}
              frameBorder="0"
              height="100%"
              src="https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0"
              title="featured-video"
              width="100%"
            />
          </section>,
        ],
        "className": "PopUp",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-times"
              />,
              <h1
                class="game-name"
              >
                Don't Be A Loser
              </h1>,
              <div
                class="game-info"
              >
                <p
                  class="age-range"
                >
                  <i
                    class="fas fa-birthday-cake"
                  />
                   Ages 
                  13
                   +
                </p>
                <p
                  class="num-of-players"
                >
                  <i
                    class="fas fa-users"
                  />
                   
                  4
                   - 
                  10
                   players
                </p>
                <p
                  class="num-of-minutes"
                >
                  <i
                    class="fas fa-clock"
                  />
                   
                  30
                   min.
                </p>
              </div>,
              <p
                class="game-description"
              >
                Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!
              </p>,
              <p
                class="difficulty-level"
              >
                Challenge level: 
                average
                <i
                  className="fas fa-thermometer-half"
                />
              </p>,
              <img
                alt="game in box"
                className="carousel-image"
                src="https://cdn.shopify.com/s/files/1/1911/5793/products/DBAL-main-photo-box_350x.jpg?v=1512623523"
              />,
            ],
            "class": "game-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-times",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Don't Be A Loser",
                "class": "game-name",
              },
              "ref": null,
              "rendered": "Don't Be A Loser",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    class="age-range"
                  >
                    <i
                      class="fas fa-birthday-cake"
                    />
                     Ages 
                    13
                     +
                  </p>,
                  <p
                    class="num-of-players"
                  >
                    <i
                      class="fas fa-users"
                    />
                     
                    4
                     - 
                    10
                     players
                  </p>,
                  <p
                    class="num-of-minutes"
                  >
                    <i
                      class="fas fa-clock"
                    />
                     
                    30
                     min.
                  </p>,
                ],
                "class": "game-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        class="fas fa-birthday-cake"
                      />,
                      " Ages ",
                      13,
                      " +",
                    ],
                    "class": "age-range",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "class": "fas fa-birthday-cake",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " Ages ",
                    13,
                    " +",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        class="fas fa-users"
                      />,
                      " ",
                      4,
                      " - ",
                      10,
                      " players",
                    ],
                    "class": "num-of-players",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "class": "fas fa-users",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " ",
                    4,
                    " - ",
                    10,
                    " players",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        class="fas fa-clock"
                      />,
                      " ",
                      "30",
                      " min.",
                    ],
                    "class": "num-of-minutes",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "class": "fas fa-clock",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " ",
                    "30",
                    " min.",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!",
                "class": "game-description",
              },
              "ref": null,
              "rendered": "Don't Be a Loser is a fun party game where you don't have to win but you just don't want to be the BIG LOSER! This game is easy to play and designed with social situations in mind, allowing players to freely leave and come back without interrupting the game!",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Challenge level: ",
                  "average",
                  <i
                    className="fas fa-thermometer-half"
                  />,
                ],
                "class": "difficulty-level",
              },
              "ref": null,
              "rendered": Array [
                "Challenge level: ",
                "average",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-thermometer-half",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "game in box",
                "className": "carousel-image",
                "src": "https://cdn.shopify.com/s/files/1/1911/5793/products/DBAL-main-photo-box_350x.jpg?v=1512623523",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <iframe
              allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
              allowFullScreen={true}
              frameBorder="0"
              height="100%"
              src="https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0"
              title="featured-video"
              width="100%"
            />,
            "class": "youtube-video",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "allow": "accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture",
              "allowFullScreen": true,
              "frameBorder": "0",
              "height": "100%",
              "src": "https://www.youtube.com/embed/ZYvvhOzYx_4?autoplay=0",
              "title": "featured-video",
              "width": "100%",
            },
            "ref": null,
            "rendered": null,
            "type": "iframe",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
